model-optimizer/extensions/middle/EltwiseInputReshape.py

```diff
@@ -65,58 +65,46 @@ def find_and_replace_pattern(self, graph: Graph):
 
 
 class EltwiseInputReshape(MiddleReplacementPattern):
-    enabled = True
-    force_clean_up = True
-
-    def run_after(self):
-        from extensions.middle.pass_separator import MiddleStart
-        return [MiddleStart]
+    # This pass should be called directly from pipeline before layout change and other permutations
+    enabled = False
 
     def find_and_replace_pattern(self, graph: Graph):
-        for node in graph.get_data_nodes():
-            # Get all requested shapes for current node
-            # This mapping will contain pairs like {shape:[list of consumers nodes]}
-            mapping = {}
-            for consumer in node.out_nodes():
-                edge_attrs = graph.get_edge_data(node.id, consumer.id)[0]
-                if 'new_shape' in edge_attrs:
-                    if np.array_equal(edge_attrs['new_shape'], node.shape):
-                        continue
-                    new_shape = tuple([x for x in edge_attrs['new_shape']])
-                    if not new_shape in mapping:
-                        mapping.update({new_shape: [consumer]})
-                    else:
-                        mapping[new_shape].append(consumer)
+        for node in graph.get_op_nodes():
+            for out_port_idx in node.out_ports():
+                mapping = {}
+                output_port = node.out_port(out_port_idx)
+                output_port_shape = output_port.data.get_shape()
+                for consumer_port in output_port.get_destinations():
+                    if consumer_port.node.has_valid('is_eltwise') and consumer_port.node['is_eltwise'] == True:
+                        consumer_output_shape = consumer_port.node.out_port(0).data.get_shape()
+                        if np.array_equal(consumer_output_shape, output_port_shape):
+                            continue
+                        # Set edge attribute new_shape for further transformation pass
+                        new_shape = output_port_shape
+                        if len(output_port_shape) != len(consumer_output_shape):
+                            for x in range(len(consumer_output_shape) - len(output_port_shape)):
+                                new_shape = np.insert(new_shape, 0, 1)
+                        new_shape = tuple([x for x in new_shape])
+                        if not new_shape in mapping:
```

Comment:
Why do you save them in the `mapping`? You can directly insert the Reshape operation when you detected the eltwise which input should be reshaped.

Reply:
The mapping is created to know for which eltwise consumers the same normalization (i.e. Reshape with output shape) is required. According to your way we can have Reshapes with duplicate output shapes that is not optimal.